src/components/Profile/index.js

```diff
@@ -19,8 +41,108 @@ class AuthButton extends Component {
+      super(props)
+      this.handleChangeName = this.handleChangeName.bind(this)
+      this.handleChangeTarget = this.handleChangeTarget.bind(this)
+      this.handleSubmit = this.handleSubmit.bind(this)
+    }
+    renderGroups() {
+      const { groups } = this.props
+      if(groups) {
+        console.log(groups.data)
+      }
+      return (
+      	<Table>
+      	  <TableBody>
+            {groups && (
+               groups.data.map((group, index) => {
+                 const groupId = group.ref["@ref"].id
+                 return (
+                   <TableRow key={index}>
+                     <MuiTableCell>
+                       <Link to={`/${groupId}`}>
+                         <Typography variant="h3">
+                           {group.data.name} - {group.data.weekly_target_km} km
+                         </Typography>
+                       </Link>
+		     </MuiTableCell>
+		     <MuiTableCell>
+                       <IconButton onClick={() => this.handleRequestDelete(groupId, group.data.name)} aria-label="delete">
+                         <DeleteIcon />
+                       </IconButton>
+		     </MuiTableCell>
+                   </TableRow>
+                 )
+               })
+            )}
+          </TableBody>
+        </Table>
+      )
+    }
+    handleRequestDelete(groupId, name) {
+      const shouldDelete = confirm(`Do you really want to delete the group ${name}? This cannot be undone`) // eslint-disable-line
+      if (shouldDelete) {
+      	this.props.user.jwt().then((jwt) =>  {
+      	  return api.deleteGroup(jwt, groupId)
+      	}).then(() => {
+          this.props.updateGroups()
+        })
+      }
+    }
+    handleChangeName(event) {
+      this.setState({newGroupName: event.target.value}) 
+    }
+    handleChangeTarget(event) {
+      this.setState({newGroupTarget: event.target.value})
+    }
+    handleSubmit() {
+      const { newGroupName, newGroupTarget } = this.state
+      event.preventDefault() // eslint-disable-line
+      this.props.user.jwt().then((jwt) =>  {
+        return api.createGroup(jwt, newGroupName, newGroupTarget)
+      }).then(() => 
+        this.props.updateGroups()
+      )
+    }
+    renderCreateGroup() {
+      if (this.props.isAuthenticated) {
+      	return (
+      	 <>
+           <Typography variant="h1">{"Create a new group"}</Typography>
+
+      	 	 <form noValidate autoComplete="off" onSubmit={this.handleSubmit}>
+      	 	   <TextField value={this.state.newGroupName} onChange={this.handleChangeName} label="name" />
+      	 	   <TargetTextField 
+                     value={this.state.newGroupTarget}
+                     onChange={this.handleChangeTarget}
+                     label="target"
+                     InputProps={{
+                       endAdornment: 
+                         <InputAdornment position="end">km</InputAdornment>}}
+                   />
+      	           <Input type="submit" value="Create Group"/>
+      	 	 </form>
```

Comment:
These states can be handled super sexy via `useState` hooks btw... and no need to bind the handlers if we declare them as function expressions. e.g. `const handleChangeName = () => {...}` Immutable and prettier ✨